packages/performance-tests/test/TransformerRegression.test.ts

```diff
@@ -0,0 +1,135 @@
+
+async function sleep(ms: number) {
+  return new Promise((resolve) => {
+    setTimeout(resolve, ms);
+  });
+}
+const testCasesMap = new Map([
+  ["identity transform", require("./identity-transformer")],
+]);
+
+const loggerCategory = "Transformer Performance Tests Identity";
+const outputDir = path.join(__dirname, ".output");
+
+const setupTestData = async () => {
+  const logLevel = +(process.env.LOG_LEVEL ?? "2");
+  if (LogLevel[logLevel] !== undefined) {
+    Logger.initializeToConsole();
+    Logger.setLevelDefault(LogLevel.Error);
+    Logger.setLevel(loggerCategory, LogLevel.Info);
+    Logger.setLevel(TransformerLoggerCategory.IModelExporter, logLevel);
+    Logger.setLevel(TransformerLoggerCategory.IModelImporter, logLevel);
+    Logger.setLevel(TransformerLoggerCategory.IModelTransformer, logLevel);
+  }
+  let usrEmail;
+  let usrPass;
+  if(process.env.V2_CHECKPOINT_USER_NAME){
+    usrEmail = process.env.V2_CHECKPOINT_USER_NAME;
+    usrPass = process.env.V2_CHECKPOINT_USER_PASSWORD;
+  } else if(process.env.V1_CHECKPOINT_USER_NAME){
+    usrEmail = process.env.V1_CHECKPOINT_USER_NAME;
+    usrPass = process.env.V1_CHECKPOINT_USER_PASSWORD;
+  } else {
+    usrEmail = process.env.IMODEL_USER_NAME;
+    usrPass = process.env.IMODEL_USER_PASSWORD;
+  }
+
+  assert(usrEmail, "user name was not configured");
+  assert(usrPass, "user password was not configured");
+  const user = {
+    email: usrEmail,
+    password: usrPass,
+  };
+
+  assert(process.env.OIDC_CLIENT_ID, "");
+  assert(process.env.OIDC_REDIRECT, "");
+  assert(process.env.IMJS_URL_PREFIX, "")
+  assert(process.env.OIDC_SCOPE, "List scopes");
+  const authClient = process.env.CI === "1"
+    ? new TestBrowserAuthorizationClient({
+      clientId: process.env.OIDC_CLIENT_ID,
+      redirectUri: process.env.OIDC_REDIRECT,
+      scope: process.env.OIDC_SCOPE,
+      authority: `https://${process.env.IMJS_URL_PREFIX}ims.bentley.com`,
+    }, user)
+    : new NodeCliAuthorizationClient({
+      clientId: process.env.OIDC_CLIENT_ID,
+      redirectUri: process.env.OIDC_REDIRECT,
+      scope: process.env.OIDC_SCOPE,
+    });
+
+  await authClient.signIn();
+
+  const hostConfig  = new IModelHostConfiguration();
+  hostConfig.authorizationClient = authClient;
+  const hubClient = new IModelsClient({ api: { baseUrl: `https://${process.env.IMJS_URL_PREFIX}api.bentley.com/imodels` } });
+  hostConfig.hubAccess = new BackendIModelsAccess(hubClient);
+  await IModelHost.startup(hostConfig);
+
+  return realizeAsyncIterator(getTestIModels());
+};
+
+// Mocha tests must know the test cases ahead time, so we collect the the Imodels first before beginning the tests
+async function realizeAsyncIterator(getTestIModels:AsyncGenerator<TestIModel>) {
+  let arrImodels:TestIModel[] = [];
+  const iModelIdStr = process.env.IMODEL_IDS;
+  assert(iModelIdStr, "no Imodel Ids")
+  const iModelIds = iModelIdStr === "*" ? "" : iModelIdStr.split(",")
```

Comment:
don't handle the filtering inside this utility, it's role isn't to filter, but to flatten an existing iterator. Instead we should filter elsewhere and pass the filtered iterator to the realize function.

In this case you should probably add a `filter` callback parameter to `getIModels` so you can avoid it making unnecessary project data fetches on imodel ids that will be ignored.

Reply:
done. I think i created a filter function that is sufficient, but don't know if its specifically what you intended. It is in TestUtils.ts